src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -141,11 +146,13 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     // this would add duplicate callbacks and would screw up the graph highlighter. If, however,
     // we are being asked to initialize a different cy instance, we assume the current one is now obsolete
     // so we do want to initialize the new cy instance.
-    if (this.graphHighlighter && this.graphHighlighter.cy === cy) {
+    if (this.cy === cy) {
```

Comment:
The first sentence of the long comment from lines 144-48 is no longer relevant due to the change. Since this new code isn't using the highlighter's cached cy but instead is using its own cached cy, I think this means just line 144 can be deleted but the rest of the comment is still relevant.

I will say, be careful caching the "cy" object. Because if it gets destroyed later, we do NOT want the cached cy to hang around and we do NOT want duplicate callbacks defined (which was happening before and why this old line 144 if-stmt is here in the first place - it was ensuring duplicate callbacks are never set). Looks like this is still correct, because the new line 149 is making sure to return if the cy object is the same as before. I just wanted to point this out. We must make sure this "cy" reference is nulled out/replaced with the new "cy" object when one is created.

Reply:
Updated the comment and made sure the cy is removed on the 'destroy' event